python/pyspark/sql/tests/test_udtf.py

```diff
@@ -2948,21 +2948,21 @@ def eval(self):
         err = "UDTF_ARROW_TYPE_CAST_ERROR"
 
         for ret_type, expected in [
-            ("x: boolean", [Row(x=True)]),
-            ("x: tinyint", [Row(x=1)]),
-            ("x: smallint", [Row(x=1)]),
-            ("x: int", [Row(x=1)]),
-            ("x: bigint", [Row(x=1)]),
+            ("x: boolean", err),
+            ("x: tinyint", err),
+            ("x: smallint", err),
+            ("x: int", err),
+            ("x: bigint", err),
```

Comment:
Is this a behavior change? I wonder if we should have a separate flag for enabling this optimization.

Reply:
It can be. But I would just land this because it is turned off by default, it was added in 3.5, and it is Spark 4.0. Strings to other type conversions don't quite make sense in any event.

Reply:
The behaviours with/without Arrow are pretty much different anyway.